openmoltools/packmol.py

```diff
@@ -157,3 +158,52 @@ def approximate_volume(pdb_filenames, n_molecules_list, box_scaleup_factor=2.0):
         volume += molecule_volume * n_molecules_list[k]
     box_size = volume**(1.0/3.0) * box_scaleup_factor
     return box_size
+
+
+def approximate_volume_by_density( smiles_strings, n_molecules_list, density = 1.0, box_scaleup_factor = 1.1):
+    """Generate an approximate box size based on the number and molecular weight of molecules present, and a target density for the final solvated mixture. If no density is specified, the target density is assumed to be 1 g/ml.
+
+    Parameters
+    ---------- 
+    smiles_strings : list(str)
+        List of smiles strings for each component of mixture.
+    n_molecules_list : list(int)
+        The number of molecules of each mixture component.
+    box_scaleup_factor : float, optional, default = 1.1
+        Factor by which the estimated box size is increased
+    density : float, optional, default 1.0
```

Comment:
Is a density of 1 g/mL really going to be universally useful?  It would seem that the better approach would be to require the user to specify the density.